### ISSUE TITLE
Training-long time waiting after each epoch

### ISSUE THREAD

**Issue by tzhong518** [HRNet/HRNet-Semantic-Segmentation#152]
Hi, thanks for your work.
I'm training hrnet_w48 with pytorch1.1.  I use 2 2080ti and the CUDA version is 10.0.

It takes long time(30 min for me) to validate the network after each epoch.  Is this normal?
I noticed the gpu usage goes to zero sometimes in this process, how could I accelerate it? 

`2020-06-17 08:25:40,211 Epoch: [60/484] Iter:[300/495], Time: 0.68, lr: 0.008866, Loss: 0.275929
2020-06-17 08:26:48,085 Epoch: [60/484] Iter:[400/495], Time: 0.68, lr: 0.008862, Loss: 0.270115
2020-06-17 08:55:40,801 => saving checkpoint to output/cityscapes/seg_hrnet_w48_train_512x1024_sgd_lr1e-2_wd5e-4_bs_12_epoch484_gpucheckpoint.pth.tar
2020-06-17 08:55:46,280 Loss: 0.330, MeanIU:  0.5560, Best_mIoU:  0.6180`

Any help is welcome! Thank you!

**Comment by tzhong518** [comment]
solved after upgrading to pytorch1.2

**Comment by GutlapalliNikhil** [comment]
Hi @tzhong518 , Found any solution for it?

**Comment by troymaple** [comment]
> Hi @tzhong518 , Found any solution for it?

I found that the most time-consuming part was the `get_confusion_matrix` function in `validate`, mainly due to the `np.argmax` (on cpu) for high-resolution prediction.

An improved function is to use the torch function on the gpu:
```
def get_confusion_matrix_gpu(label, pred, size, num_class, ignore=-1):
    seg_pred = torch.argmax(pred, dim=1)
    seg_gt = label

    ignore_index = seg_gt != ignore
    seg_gt = seg_gt[ignore_index]
    seg_pred = seg_pred[ignore_index]


    index = seg_gt * num_class + seg_pred
    label_count = torch.bincount(index, minlength=num_class**2)
    confusion_matrix = label_count.reshape((num_class, num_class))
    return confusion_matrix.cpu().numpy()
```
